lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -279,6 +279,10 @@ class AwsCompileFunctions {
       delete newFunction.Properties.VpcConfig;
     }
 
+    if (functionObject.concurrency && typeof functionObject.concurrency === 'number') {
```

Comment:
Use lodash here: `if (_.isInteger(functionObject.concurrency)) { ...`

That's easier, disallows floats and you do not need to check for nil values explicitly.

Reply:
Maybe you could print a warning, if it is there, but not an integer? Would be a nice improvement.

Reply:
Good idea! I will use lodash and print warning.